JWT/JWT.cs

```diff
@@ -137,34 +139,38 @@ public static string Decode(string token, byte[] key, bool verify = true)
                 var decodedCrypto = Convert.ToBase64String(crypto);
                 var decodedSignature = Convert.ToBase64String(signature);
 
-                if (decodedCrypto != decodedSignature)
-                {
-                    throw new SignatureVerificationException(string.Format("Invalid signature. Expected {0} got {1}", decodedCrypto, decodedSignature));
-                }
-
-                // verify exp claim https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.4
-                var payloadData = JsonSerializer.Deserialize<Dictionary<string, object>>(payloadJson);
-                if (payloadData.ContainsKey("exp") && payloadData["exp"] != null)
-                {
-                    // safely unpack a boxed int 
-                    int exp;
-                    try { exp = Convert.ToInt32(payloadData["exp"]); }
-                    catch (Exception) 
-                    { 
-                        throw new SignatureVerificationException("Claim 'exp' must be an integer.");
-                    }
-
-                    var secondsSinceEpoch = Math.Round((DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds);
-                    if (secondsSinceEpoch >= exp)
-                    {
-                        throw new SignatureVerificationException("Token has expired.");
-                    }
-                }
+                Verify(decodedCrypto, decodedSignature, payloadJson);
             }
 
             return payloadJson;
         }
 
+        private static void Verify(string decodedCrypto, string decodedSignature, string payloadJson)
+        {
+            if (decodedCrypto != decodedSignature)
+            {
+                throw new SignatureVerificationException(string.Format("Invalid signature. Expected {0} got {1}", decodedCrypto, decodedSignature));
+            }
+
+            // verify exp claim https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-4.1.4
+            var payloadData = JsonSerializer.Deserialize<Dictionary<string, object>>(payloadJson);
+            object value;
+            if (payloadData.TryGetValue("exp", out value) && value != null)
+            {
+                // safely unpack a boxed int 
+                if (!(value is int))
+                {
+                    throw new SignatureVerificationException("Claim 'exp' must be an integer.");
+                }
+
+                var secondsSinceEpoch = Math.Round((DateTime.UtcNow - UnixEpoch).TotalSeconds);
+                if (secondsSinceEpoch >= (int)value)

```

Comment:
I think this code may have been updated since you originally started this PR so that now it does Convert.ToInt32() which is more permissive than "is int" and "(int)value".  I think Convert.ToInt32() is probably better, since it'll handle longs and whatnot.